src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -4026,6 +4026,146 @@ private function matches(): bool {
 		return true;
 	}
 
+	/**
+	 * Get the doctype name from a doctype token.
+	 *
+	 * @since 6.7.0
+	 *
+	 * @return string|null The doctype name, or null if not a doctype token.
+	 */
+	public function get_doctype_name(): ?string {
+		$doctype = $this->parse_doctype();
+
+		if ( null === $doctype ) {
+			return null;
+		}
+
+		return $doctype[0];
+	}
+
+	public function parse_doctype(): ?array {
```

Comment:
I think this should be protected, not public. We can set doctype details on processor state and add getters for them. Then we can do this parsing just once when the doctype is reached.

Maybe there's a better place for this parsing to happen.